src/frequenz/sdk/timeseries/logical_meter/_formula_generators.py

```diff
@@ -0,0 +1,93 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Generators for formulas from component graphs."""
+
+from frequenz.channels import Sender
+
+from ...actor import ChannelRegistry, ComponentMetricRequest
+from ...microgrid import ComponentGraph
+from ...microgrid.component import ComponentCategory, ComponentMetricId
+from ._formula_engine import FormulaEngine
+from ._resampled_formula_builder import ResampledFormulaBuilder
+
+
+class FormulaGenerator:
+    """A class for generating formulas from the component graph."""
+
+    def __init__(
+        self,
+        namespace: str,
```

Comment:
Maybe `request_namespace` or something that makes it more obvious what is referring to? Not sure, just a suggestion.

Reply:
I think we need to qualify it only when we move out of the data pipeline.  It has a shared meaning in the data pipeline code,  so I'd say this is fine.